python/ql/src/experimental/Security-new-dataflow/promote.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -Eeuo pipefail # see https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/
+
+# Promotes new dataflow queries to be the real ones
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+cd $SCRIPTDIR
+for file in $(find . -mindepth 2); do
+    echo "Promoting $file"
+    mkdir -p "../../Security/$(dirname $file)"
```

Comment:
So, just to check, since this says `Security` and not `Security-new-dataflow` I take it we'll be doing the latter promotion manually, and this is more a useful tool for when `Security` becomes the staging area for new queries?

I'm also wondering: what if we end up with a poorly performing query in `Security` -- do we want to continually test this (as it'll get promoted alongside everything else) even though we know it'll e.g. time out on a large number of projects during a dist upgrade?

Reply:
This is just a helper script for when you want to create a testing branch for a dist-compare/dist-upgrade run. It replaces the existing security queries with the new queries. (it's easier to run a script than manually moving/copying files).

Since you would still need to commit the changes, if there are some queries you _know_ shouldn't be part of your dist-compare/dist-upgrade, you can just not commit those :blush: